Tests/GitHubCodespaces.tests.ps1

```diff
@@ -186,6 +253,43 @@ Describe 'GitHubCodespaces\Get-GitHubCodespace' {
     }
 }
 
+Describe 'GitHubCodespaces\Get-GitHubCodespaceMachine' {
+    BeforeAll {
+        # Suppress HTTP 202 warning for codespace creation
+        $WarningPreference = 'SilentlyContinue'
+
+        $newGitHubCodespaceParms = @{
+            OwnerName = $repo.owner.login
+            RepositoryName = $defaultRepositoryName
+        }
+        $null = New-GitHubCodespace @newGitHubCodespaceParms -Wait
+    }
+
+    Context 'When listing machine types a codespace can transition to' {
+        BeforeAll {
+            $codespaceMachine = Get-GitHubCodespace @newGitHubCodespaceParms | Select-Object -First 1 | Get-GitHubCodespaceMachine
+        }
+
+        It 'Should return an object of the correct type' {
+            $codespaceMachine | Should -BeOfType PSCustomObject
```

Comment:
Since I don't see any existing tests that validate a GitHub.* TypeName, I'm assuming you are looking for this: 4bfd42228409510e0407780efd5a363b6b2d762b rather than changing all the existing test criteria.  If I've misunderstood, then please clarify.

Reply:
Most existing tests validate the output typename.  Here are a few examples:

https://github.com/microsoft/PowerShellForGitHub/blob/5e1a4b312996ac894e01fa13c9677b1f302becfb/Tests/GitHubLabels.tests.ps1#L101-L111

https://github.com/microsoft/PowerShellForGitHub/blob/5e1a4b312996ac894e01fa13c9677b1f302becfb/Tests/GitHubPullRequests.tests.ps1#L35-L50

https://github.com/microsoft/PowerShellForGitHub/blob/5e1a4b312996ac894e01fa13c9677b1f302becfb/Tests/GitHubRepositoryForks.tests.ps1#L46-L49

Reply:
Changed